### ISSUE TITLE
Will the GitHub get an sounds examples/library folder?

### ISSUE THREAD

**Issue by bas080** [ccrma/chuck#25]
During my creative process I like to listen to many diffrent sounds others made. 

It would be nice to have an open and easy way to share user creations so others can review it and use/improve it. Other options could be a new repository or a branch. My preference goes to a repository as it is easier for users to find and use. 

Btw, loving ChucK. Looking forward to making many sounds.

**Comment by markcerqueira** [comment]
Another option might be SoundCloud or something like that. People can upload their works and link to the project behind that work. We can then curate a ChucK playlist and you can listen without needing to check out a repository.

**Comment by bas080** [comment]
Who is we?

What I am proposing is that the we is everyone that knows how to use github and writes ChucK code.

**Comment by aknuds1** [comment]
This is kind of the motivation for my [ChucK Demos](http://chuckdemos.com/) project. It's ChucK running within the browser (i.e. as JavaScript), and I'm accepting contributions. Problem right now is that my ChucKJS port is very incomplete, but that will change once my switch to automatic porting through Emscripten is complete. Work on the latter goes on in my [ChucK fork](https://github.com/aknuds1/chuck). Feel free to help out if you like :)

**Comment by bas080** [comment]
@aknuds1, good project. I am sure it will improve sharing, collaboration and the web. I'll keep an eye on your project. Till then I will be waiting for a good platform to collaborate on ChucK instruments/examples.

**Comment by markcerqueira** [comment]
@bas080 - I'm not sure how SoundCloud playlists work. If people can just contribute to a playlist, we can just share the playlist and tell people to add. If it requires some sort of moderation/curation, we can have any community members interested in curation handle it.

I agree with you in that it would be cool to have a big example folder of instruments, but I don't think hosting a library of sound files in GitHub is the best way to expose people to all the sounds produced by those instruments. 

Cheers!

**Comment by spencersalazar** [comment]
I would be down to create a new repository under the ccrma/ organization for pulling in user examples/demos. I think its important to both listen and see code. What would be super cool is to have a server automatically generate an audio file for new commits to the example repo. This could go up in SoundCloud or elsewhere. Its also very easy in miniAudicle to export sound files- its a menu item under File->Export, so I don't really see why not have both if someone is already going through the trouble of submitting an example for public browsing.

**Comment by markcerqueira** [comment]
@spencersalazar - If we do go that route, we can automate with Jenkins, GitHub hooks, and the SoundCloud API to auto-upload stuff.

**Comment by bas080** [comment]
@spencersalazar, automatically generated would be cool for people that want to use the .wav for other projects. What I did to listen to the files was to write a Bash script that loops over certain .ck files and plays them one at a time for 3 seconds. I checkout the code of the ones that get my attention. This makes it easy for me to explore the Chucks.

**Comment by hughrawlinson** [comment]
@markcerqueira I built that, it's [here](https://github.com/hughrawlinson/chuck_demos). Use/modify it if you like!

**Comment by spencersalazar** [comment]
> This repo uses my own ChucK debian package

@hughrawlinson would you be interested in making this public, maybe as a PPA on Launchpad? We have been trying to make this work (especially nice for ARM/Raspberry Pi) for years but the APT/.deb system has proven to be too mystifying.

**Comment by hughrawlinson** [comment]
@spencersalazar I'd be happy to, I'll start on making it happen and put any updates in new thread 😄
